engine/runtime-test-instruments/src/main/java/org/enso/interpreter/test/instruments/CodeIdsTestInstrument.java

```diff
@@ -21,8 +21,24 @@
     services = CodeIdsTestInstrument.class)
 public class CodeIdsTestInstrument extends TruffleInstrument {
   public static final String INSTRUMENT_ID = "ids-test";
+  private static final Class<?> exprNodeClass;
+  private static final Method exprNodeGetId;
+  private static final Class<?> tailCallExceptionClass;
   private Env env;
 
+  /**
+   * Use reflection to call methods from `runtime` project. We cannot use `runtime` as a dependency.
+   */
+  static {
+    try {
+      exprNodeClass = Class.forName("org.enso.interpreter.node.ExpressionNode");
+      exprNodeGetId = exprNodeClass.getMethod("getId");
```

Comment:
Create a service class:
```java
interface GetId {
   UUID getId(Node n);
}
```
and `require service` it in `module info`. Obtain its implementation via `ServiceLoader`. Register its implementation in `runtime-xyz/test` via `@ServiceProvider` annotation.

I can't guarantee it is going to work 100% this way, but it may work fine (btw. use `org.openide.util.Lookup` instead of `ServiceLoader`) if you hit problems between modules/unnamed modules.

Reply:
The `GetId` interface may be found by [Lookups.metaInfServices](https://github.com/enso-org/enso/blob/857e87430d59a01ef30017a88b892940f73964e9/lib/java/persistance/src/main/java/org/enso/persist/PerMap.java#L18).

Reply:
Done in b98526ee882e4c10ce56d90b6750fd6e3b1754a1